### PR TITLE
Bump Eclipse version and require Maven 3.9.0

### PR DESCRIPTION
**Pull request by ctubbsii**

### CODE REVIEW

**Comment by ctubbsii** [comment]
I tried to fix more things by adding dependency convergence, but I couldn't figure out where the dependency versions were coming from. I can hard-code in a bunch of them to fix their versions and force dependency resolution, but it would be fragile if dependencies changed in the future. As is, though, a lot of the complexity needed in the formatter-maven-plugin to get dependency convergence is really coming from this library, because it doesn't have dependency convergence of its own.

Related question: is upstream Eclipse ever going to publish jsdt-core themselves? It'd be nice to stop packaging a custom version for use with formatter-maven-plugin.

**Comment by hazendaz** [comment]
This module is for javascript.  Its smaller than it was but doesn't solve issue here https://github.com/revelc/jsdt-core/pull/50.

Not to be confused with the java code one.  AFAIK, eclipse still does not produce this for central.  This is using the OSGI p2 repository to pull the artifacts since they are not in maven.  Eventually would like to solve the issue above so we have a good balance on what this repo needs vs other users using it.  If we were to try modular on this project, this is a current show stopper as a lot of overlap.

The dependency convergence issue if I'm not mistaken was the java one as Eclipse was shipping that with version ranges which are now fixed since 2022.  This one doesn't have any version ranges listed to it.

It is a bit of foo magic.  I don't care for OSGI and barely understand its purpose.  It took a lot of testing adjusting that file until our builds worked well.  I tried adding some back per the open ticket but had not gotten back to it recently.  I don't think fixing that is a rush.  It is much smaller in general so that helps improve formatter build time when loading the m2 cache.

File: pom.xml [hunk 1]
```diff
@@ -60,6 +60,7 @@ SPDX-License-Identifier: EPL-2.0
     <mdep.analyze.skip>true</mdep.analyze.skip>
     <!-- rat plugin is no good at validating EPL headers out of the box -->
     <rat.skip>true</rat.skip>
+    <revelc.min-build-mvn>3.9.0</revelc.min-build-mvn>
```
Comment (hazendaz) [inline pom.xml:63]: Why not just require 3.9.8?  3.9.0 is pretty buggy, so was the first few.  3.9.8 is best stable.

**Comment by hazendaz** [comment]
LGTM other than comment on why not maven 3.9.8.  Feel free to merge away :)

File: pom.xml (hunk 1 above)
Comment (ctubbsii) [reply pom.xml:63]: This is just the minimum. We can recommend newer. This minimum was derived automatically from the plugins configured in the pom.xml... they require at least 3.9.0.

**Pull request merged**